hydra-node/json-schemas/api.yaml

```diff
@@ -620,285 +391,580 @@ components:
+        and the given participant identifiers (commonly public key hashes), but
+        we are not part of it. It could denote a misconfiguration, or simply
+        some other group of parties opening a head.
+      payload:
+        $ref: "api.yaml#/components/schemas/IgnoredHeadInitializing"
+
+    # END OF SERVER OUTPUT MESSAGES
+
+    DraftCommitTxRequest:
+      title: DraftCommitTxRequest
+      description: |
+        Provide utxos that will be used to draft a commit transaction that will be sent back to the user.
+        Script utxos must provide a witness, regulars can omit this.
+      payload:
+        $ref: "api.yaml#/components/schemas/UTxOWithWitness"
+        example:
+          {
+            "0406060506030602040508060506060306050406020207000508040704040203#89": {
+              "address": "addr_test1vz66ue36465w2qq40005h2hadad6pnjht8mu6sgplsfj74q9pm4f4",
+              "value": {
+                "lovelace": 7620669
+              }
+            },
+            "6f066e0f6ba373c0ea7d8b47aefd7e14d1a781698cd052d0254afe65e039b083#0": {
+              "address": "addr_test1wqv4z4hc0u5e2c3sppfdu8ckn82hfegpkjagsm4t8ttvlycg9mkca",
+              "datum": null,
+              "datumhash": "bb30a42c1e62f0afda5f0a4e8a562f7a13a24cea00ee81917b86b89e801314aa",
+              "inlineDatum": null,
+              "referenceScript": null,
+              "value": {
+                "lovelace": 1034400
+              },
+              "witness": {
+                "datum": "02",
+                "plutusV2Script": {
+                  "cborHex": "484701000022200101",
+                  "description": "",
+                  "type": "PlutusScriptV2"
+                },
+                "redeemer": "01"
+              }
+            },
+            "87a0c1e14be2cd8c385b6fe5a40b024b7201da9df375542029d91ccaba01ac82#0": {
+                "address": "addr_test1wqv4z4hc0u5e2c3sppfdu8ckn82hfegpkjagsm4t8ttvlycg9mkca",
+                "datum": null,
+                "inlineDatum": {
+                  "int": 2
+                },
+                "inlineDatumhash": "bb30a42c1e62f0afda5f0a4e8a562f7a13a24cea00ee81917b86b89e801314aa",
+                "referenceScript": null,
+                "value": {
+                  "lovelace": 905100
+                },
+                "witness": {
+                  "datum": null,
+                  "plutusV2Script": {
+                    "cborHex": "484701000022200101",
+                    "description": "",
+                    "type": "PlutusScriptV2"
+                  },
+                  "redeemer": "01"
+                }
+              }
+          }
+
+    DraftCommitTxResponse:
+      title: DraftCommitTxResponse
+      description: |
+          Emitted by the server after drafting a commit transaction with the user provided utxos. Transaction returned to the user is in it's cbor representation encoded as Base16.
+      payload:
+        $ref: "api.yaml#/components/schemas/Transaction"
```

Comment:
I'm not sure if you realized this, but the response used here is generated **NOT** using the `ToJSON Tx` type, but instead using [serialiseToTextEnvelope](https://github.com/input-output-hk/hydra/blob/66c431f94985274626b82d949a66e820db3b432b/hydra-node/src/Hydra/API/HTTPServer.hs#L58).

The response produces matches the schema (because it is very lax): e.g. I received this when trying it on the demo:

```
{"cborHex":"84a70082825820dfb4306f47f1126d272aac067a498cd7832f2a62cc394cfe83fd5c7a8126d09601825820dfb4306f47f1126d272aac067a498cd7832f2a62cc394cfe83fd5c7a8126d096020d81825820dfb4306f47f1126d272aac067a498cd7832f2a62cc394cfe83fd5c7a8126d096021281825820018c5e5c4e68e24a7d82b0103926c1c3d855361e600e78b9d4a165cfe8fe136f000182a300581d707cb20fa71eb4c563ca283566ebe0aa65859d96c3f8cba35c52c181fd01821a00164b62a1581c1e28fdee5c0032f78736ff9d9c7a2dd05b3a59d5cd3f9724c6853221a1581cf8a68cd18e59a6ace848155a0e967af64f4d00cf8acee8adc95a6b0d01028201d8185845d8799f5820b37aabd81024c043f53a069c91e51a5b52e4ea399ae17ee1fe3cb9c44db707eb80581c1e28fdee5c0032f78736ff9d9c7a2dd05b3a59d5cd3f9724c6853221ff82581d60f8a68cd18e59a6ace848155a0e967af64f4d00cf8acee8adc95a6b0d1a012f5066021a0035d8600e81581cf8a68cd18e59a6ace848155a0e967af64f4d00cf8acee8adc95a6b0d0b582097c16109a0c9971830dc0bda51a94c5c0efbe12d1f58224c622dd2d0d18671c9a20081825820eb94e8236e2099357fa499bfbc415968691573f25ec77435b7949f5fdfaa5da058406980752bd9de792a700e3ae68e4b61aa84c93dabcfaa54c92d10f25e20c153752146080523ae4b5528e0e73db65d7bd8dc38273c812d751f7c87558212e815080581840000d87a9f80ff821a00d59f801b00000002540be400f5f6","description":"Hydra commit transaction","type":"Tx BabbageEra"}
```

But I thought I would point that out. Not sure yet whether we would expect a `txId` field on the response though.